tests/src/Fixtures/TripalImporter/TestTripalImporter.php

```diff
@@ -0,0 +1,365 @@
+    $input_type = 'REPLACE WITH VALIDATOR INPUT TYPE - @input_types';
+    $instance = $this->service_validatorPluginManager->createInstance($id);
+    //
+    // Use relevant setters to set some values.
+    // For example:
+    // $instance->setExpectedColumns($num_columns, TRUE);
+    // $instance->setFileMimeType($file_mime_type);
+    // $instance->setHeaders($this->headers);
+    // $instance->setIndices(1);
+    //
+    // Register the validator instance.
+    $validators[$input_type][$id] = $instance;
+
+    return $validators;
+  }
+
+  /**
+   * {@inheritDoc}
+   *
+   * THIS METHOD IS REQUIRED!
+   */
+  public function processValidationMessages($failures) {
+
+    $messages = [];
+    // SEE src/Plugins/Validators for VALIDATOR IDs.
+    //
+    // PROCESS MESSAGE TEMPLATE:
+    // 1: Create a $messages entry.
+    $id = 'REPLACE WITH VALIDATOR ID ANNOTATION - @id';
+    $messages[$id] = [
+      'title' => 'REPLACE WITH A MESSAGE TITLE',
+      'status' => 'todo',
+      'details' => '',
+    ];
+
+    // 2: Inspect validator-specific $failures and switch status accordingly.
+    if (array_key_exists($id, $failures)) {
+      if (!empty($failures[$id])) {
+        $messages[$id]['status'] = 'fail';
+        $messages[$id]['details'] = 'USE VALIDATOR MESSAGE PROCESSOR HERE';
+        // For example using valid_delimiter_file validator message processsor:
+        // ... = ValidDelimitedFile::processValidDelimitedFileFailures($failures[$id]);.
+      }
+      else {
+        $messages[$id]['status'] = 'pass';
+      }
+    }
+
+    return $messages;
+  }
+
+  /**
+   * {@inheritdoc}
+   *
+   * THIS METHOD IS REQUIRED!
+   */
+  public function formValidate($form, &$form_state) {
+
+    $form_values = $form_state->getValues();
+    $file_id = $form_values['file_upload'];
+
+    // Load our file object.
+    $file = $this->service_entityTypeManager->getStorage('file')->load($file_id);
+
+    // Get the mime type which is used to validate the file and split the rows.
+    $file_mime_type = $file->getMimeType();
+
+    // Configure the validators.
+    $validators = $this->configureValidators($form_values, $file_mime_type);
+    $validator_key = array_keys($validators)[0];
+
+    // A Flag to keep track if any validator fails.
+    $failed_validator = FALSE;
+
+    // Holds failed items.
+    $failures = [];
+
+    // Data validation.
+    if ($failed_validator === FALSE) {
+      $file_uri = $file->getFileUri();
+      $handle = fopen($file_uri, 'r');
+      $line_no = 0;
+
+      while (!feof($handle)) {
+        $line = fgets($handle);
+        $line_no++;
+        if (empty(trim($line))) {
+          continue;
+        }
+
+        foreach ($validators[$validator_key] as $validator_name => $validator) {
+          if (!array_key_exists($validator_name, $failures)) {
+            $failures[$validator_name] = [];
+          }
+
+          // CALL THE VALIDATOR VALIDATE METHOD WITH THE REQUIRED PARAMETERS.
```

Comment:
Expand to indicate where to call the row level validators. We should also add a block above the file opening to indicate where to call a metadata or file level validator.

NOTE: I would add a matching dotted line under the example code to make it a block and indicating not to change below this point.

```suggestion
          // CALL VALIDATE (ROW LEVEL VALIDATORS)
          // ..................................................
          // CALL THE VALIDATOR VALIDATE METHOD WITH THE REQUIRED PARAMETERS.
          // NOTE: if you are testing a raw row validator then you can call
          // it with `$line` directly. If not, the call the split row helper
          // first by uncommenting the following line.
          // $row = ImportValidationHelper::splitRowIntoColumns($line, $file_mime_type);
```